mono/metadata/threadpool-ms.c

```diff
@@ -227,9 +227,11 @@ rand_create (void)
 static guint32
 rand_next (gpointer *handle, guint32 min, guint32 max)
 {
+	MonoError error;
 	guint32 val;
-	if (!mono_rand_try_get_uint32 (handle, &val, min, max)) {
+	if (!mono_rand_try_get_uint32 (handle, &val, min, max, &error)) {
 		// FIXME handle error
+		mono_error_cleanup (&error);
 		g_assert_not_reached ();
```

Comment:
You could replace that by `mono_error_assert_ok(&error)`, so we know what was the error. It's going to assert anyway.